homeassistant/components/amcrest/__init__.py

```diff
@@ -117,31 +131,90 @@ def _has_unique_names(devices):
 
 def setup(hass, config):
     """Set up the Amcrest IP Camera component."""
-    hass.data.setdefault(DATA_AMCREST, {'devices': {}, 'cameras': []})
-    devices = config[DOMAIN]
-
-    for device in devices:
+    hass.data.setdefault(DATA_AMCREST, {DEVICES: {}, CAMERAS: []})
+
+    # pylint: disable=too-many-ancestors
+    class AmcrestChecker(Http):
```

Comment:
Please define this outside of the setup method.

Reply:
Done.